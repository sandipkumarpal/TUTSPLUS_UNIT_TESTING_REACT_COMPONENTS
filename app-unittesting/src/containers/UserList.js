import React from 'react';
import User from '../components/User';

class UserList extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            filter: ''
        }
    }
    render() {
        const {users, filter } = this.props;
        return (
            <ul className="list-group">
                {users
                    .filter(user => user.name.toLowerCase().indexOf(filter.toLowerCase()) > -1)
                    .map( user => <User key={user.name} user={user}/> )
                }
            </ul>
        );
    }
}

export default UserList;
