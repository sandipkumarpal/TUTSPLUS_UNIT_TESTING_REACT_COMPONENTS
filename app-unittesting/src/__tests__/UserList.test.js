import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow, render} from 'enzyme';
import UserList from '../containers/UserList';

const userData = [
    {name: 'Sandip'},
    {name: 'Rayel'},
    {name: 'Gopal'},
    {name: 'Akash'},
    {name: 'Monoj'}
  ];
const filterData = '';

test('<User /> render correctly', () => {
  const tree = renderer
    .create(<UserList users={userData} filter={filterData}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
