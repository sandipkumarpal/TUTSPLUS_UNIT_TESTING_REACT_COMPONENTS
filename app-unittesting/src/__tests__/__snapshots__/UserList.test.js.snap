// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> render correctly 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item"
  >
    Sandip
  </li>
  <li
    className="list-group-item"
  >
    Rayel
  </li>
  <li
    className="list-group-item"
  >
    Gopal
  </li>
  <li
    className="list-group-item"
  >
    Akash
  </li>
  <li
    className="list-group-item"
  >
    Monoj
  </li>
</ul>
`;
